js/books.js

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable max-classes-per-file */
+class StorageUtil {
+  localStorageAvailable() {
+    this.storage = null;
+
+    try {
+      this.storage = window.localStorage;
+      const x = '__storage_test__';
+      this.storage.setItem(x, x);
+      this.storage.removeItem(x);
+      return true;
+    } catch (e) {
+      return e instanceof DOMException && (
+        // everything except Firefox
+        e.code === 22
+        // Firefox
+        || e.code === 1014
+        // test name field too, because code might not be present
+        // everything except Firefox
+        || e.name === 'QuotaExceededError'
+        // Firefox
+        || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+        // acknowledge QuotaExceededError only if there's something already stored
+        && (this.storage && this.storage.length !== 0);
+    }
+  }
+
+  setItem(name, item) {
+    if (this.localStorageAvailable()) {
+      localStorage.setItem(name, item);
+    }
+  }
+
+  getItem(name) {
+    if (this.localStorageAvailable()) {
+      return localStorage.getItem(name);
+    }
+    return null;
+  }
+}
+
+class Book {
+  constructor(id, title, author) {
+    this.id = id;
+    this.title = title;
+    this.author = author;
+  }
+}
+
+class Library {
+  constructor() {
+    this.books = [];
+    this.storageUtil = new StorageUtil();
+    this.init();
+  }
+
+  init() {
+    this.books = JSON.parse(this.storageUtil.getItem('books')) || [];
+  }
+
+  save() {
+    this.storageUtil.setItem('books', JSON.stringify(this.books));
+  }
+
+  addBook(title, author) {
+    const id = this.books.length + 1;
+    const book = new Book(id, title, author);
+
+    this.books.push(book);
+    this.save();
+  }
+
+  removeBook(bookId) {
+    this.books = this.books.filter((book) => book.id != bookId); /* eslint-disable-line eqeqeq */
```

Comment:
- [ ] Please avoid disabling linters. Please always try to find the solution for the particular linter error online before defaulting to disabling the linters. In this case, you could have simply used `!==` instread.

Reply:
`!==` broke the filtering behavior. That's why the disabling line is there. We didn't do it on a whim.

Reply:
Good, but next time try to find a workaround. If `!==` breaks the filter, try researching why the variable types behave differently on JavaScript, and then maybe change the car types to make your filter comply with linter requirements. Cheers and happy coding.

Reply:
Here's an article you might find useful. https://www.google.com/amp/s/www.geeksforgeeks.org/difference-between-and-operator-in-javascript-2/amp/

My guess is your problem here was comparing a string to a numeric value. You could have converted the textual number to integer, or vice versa, and that would have allowed you to use `!==` here.